cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/ProposeNewConstitutionSPO.hs

```diff
@@ -0,0 +1,248 @@
+
+      spoColdSkeyFp :: Int -> FilePath
+      spoColdSkeyFp n = tempAbsPath' </> "pools-keys" </> "pool" <> show n </> "cold.skey"
+
+  -- Create constitution proposal
+  void $ H.execCli' execConfig
+    [ "conway", "governance", "action", "create-constitution"
+    , "--testnet"
+    , "--governance-action-deposit", show @Int 0 -- TODO: Get this from the node
+    , "--deposit-return-stake-verification-key-file", stakeVkeyFp
+    , "--anchor-url", "https://tinyurl.com/3wrwb2as"
+    , "--anchor-data-hash", proposalAnchorDataHash
+    , "--constitution-url", "https://tinyurl.com/2pahcy6z"
+    , "--constitution-hash", constitutionHash
+    , "--out-file", constitutionActionFp
+    ]
+
+  txbodyFp <- H.note $ work </> "tx.body"
+  txbodySignedFp <- H.note $ work </> "tx.body.signed"
+
+  txin1 <- queryAnyUtxo . paymentKeyInfoAddr $ head wallets
+
+  void $ H.execCli' execConfig
+    [ "conway", "transaction", "build"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--tx-in", Text.unpack $ renderTxIn txin1
+    , "--change-address", Text.unpack $ paymentKeyInfoAddr $ head wallets
+    , "--proposal-file", constitutionActionFp
+    , "--out-file", txbodyFp
+    ]
+
+  void $ H.execCli' execConfig
+    [ "conway", "transaction", "sign"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--tx-body-file", txbodyFp
+    , "--signing-key-file", paymentSKey $ paymentKeyInfoPair $ head wallets
+    , "--out-file", txbodySignedFp
+    ]
+
+  void $ H.execCli' execConfig
+    [ "conway", "transaction", "submit"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--tx-file", txbodySignedFp
+    ]
+
+  txidString <- mconcat . lines <$> H.execCli' execConfig
+    [ "transaction", "txid"
+    , "--tx-file", txbodySignedFp
+    ]
+  !propSubmittedResult
+    <- runExceptT $ handleIOExceptT IOE
+                  $ runExceptT $ foldBlocks
+                      (File $ configurationFile testnetRuntime)
+                      (File socketPath)
+                      FullValidation
+                      Nothing -- Initial accumulator state
+                      (PNC.foldBlocksCheckProposalWasSubmitted (fromString txidString))
+
+  newProposalEvents <- case propSubmittedResult of
+                        Left (IOE e) ->
+                          H.failMessage callStack
+                            $ "foldBlocksCheckProposalWasSubmitted failed with: " <> show e
+                        Right (Left e) ->
+                          H.failMessage callStack
+                            $ "foldBlocksCheckProposalWasSubmitted failed with: " <> Text.unpack (renderFoldBlocksError e)
+                        Right (Right events) -> return events
+
+  governanceActionIndex <- PNC.retrieveGovernanceActionIndex newProposalEvents
+
+  let voteFp :: Int -> FilePath
+      voteFp n = work </> gov </> "vote-" <> show n
+
+  -- Proposal was successfully submitted, now we vote on the proposal and confirm it was ratified
+  forM_ [1..3] $ \n -> do
```

Comment:
you can use `H.forConcurrently_` and save those 2 seconds on the test execution time 😄

Reply:
Saving 2 seconds is actually something like a 4% increase, I take it :heart_eyes_cat: